azure-pipelines.yaml

```diff
@@ -0,0 +1,86 @@
+# Starter pipeline
+# Build a general Node.js project with npm.
+# Add steps that analyze code, save build artifacts, deploy, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/languages/javascript
+
+name: $(Date:yyyyMMdd)$(Rev:.r)
+
+trigger:
+  - main
+  - features/*
+  - feature/*
+
+variables:
+  isMaster: $[eq(variables['Build.SourceBranch'], 'refs/heads/master')]
+  isPullRequest: $[eq(variables['Build.Reason'], 'PullRequest')]
+  branchSource: $[replace(variables['Build.SourceBranch'], 'refs/heads/', '')]
+  branch: $[replace(variables['branchSource'], '/', '-')]
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+steps:
```

Comment:
- May be we can group steps into seperate jobs `Build` and `PublishArtifact` for better organisation
- We can put `dependsOn` to tei dependencies between jobs more explicitly.

eg:
<img width="220" alt="image" src="https://github.com/DEFRA/ncea-frontend/assets/131792244/b9701146-1d69-439c-8b04-909fc574632b">

Reply:
There is known issue in copying files between stages/jobs.. Hence we are keeping build and publish in same Job